From 47f21cec44ced42ae073f314da08120989ec65fe Mon Sep 17 00:00:00 2001
From: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
Date: Thu, 25 Oct 2018 07:09:20 +0900
Subject: [PATCH 5/5] docs: add DELETE method for exiting spp_nfv

To support `exit` of `sec` command of `spp.py`, add DELETE method. It
is called with client ID as following example.

  $ curl -X DELETE http://127.0.0.1:7777/v1/nfvs/1

Signed-off-by: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
---
 docs/guides/spp-ctl/api-reference.rst | 43 +++++++++++++++++++++++++++++++++++
 1 file changed, 43 insertions(+)

diff --git a/docs/guides/spp-ctl/api-reference.rst b/docs/guides/spp-ctl/api-reference.rst
index 3ae90fd..f0661c8 100644
--- a/docs/guides/spp-ctl/api-reference.rst
+++ b/docs/guides/spp-ctl/api-reference.rst
@@ -582,6 +582,49 @@ Equivalent CLI command
     sec {client_id};patch reset
 
 
+DELETE /v1/nfvs/{client_id}
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Terminate ``spp_nfv`` or ``spp_vm`` process.
+
+* Normal response codes: 204
+* Error response codes: 400, 404
+
+Request(path)
+^^^^^^^^^^^^^
+
+.. _table_spp_ctl_nfvs_delete:
+
+.. table:: Request parameter for terminating spp_nfv or spp_vm.
+
+    +-----------+---------+-------------------------------------+
+    | Name      | Type    | Description                         |
+    |           |         |                                     |
+    +===========+=========+=====================================+
+    | client_id | integer | client id.                          |
+    +-----------+---------+-------------------------------------+
+
+Request example
+^^^^^^^^^^^^^^^
+
+.. code-block:: console
+
+    curl -X DELETE -H 'application/json' \
+    http://127.0.0.1:7777/v1/nfvs/1
+
+Response example
+^^^^^^^^^^^^^^^^
+
+There is no body content for the response of a successful ``DELETE`` request.
+
+Equivalent CLI command
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: none
+
+    sec {client_id}; exit
+
+
 API for spp_vf
 --------------
 
-- 
2.7.4

